### DISCUSSION TITLE
Caps, Numpad and Scroll locks

### DISCUSSION THREAD

**Discussion by SuperNinja-4965** [hathach/tinyusb#748] [Q&A] [answered]
This might seem like a stupid question but how do i get the status of the 3 locks on a keyboard??

**Comment by chegewara** [comment, marked as answer]
It all depends what is your use case. When you are building a keyboard device, then after connecting to host you should get 1 or 2 bytes, dont remember, with report about caps lock and num lock, also the same report will be received when you change caps/num lock on second keyboard.

**Reply by SuperNinja-4965** [reply]
well i am using it to create a macro keypad. i just need to be able to periodically get the status of the caps lock keys etc to see if caps lock is on or off

**Reply by SuperNinja-4965** [reply]
This is hte repo for the code where i am trying to do this: 
https://github.com/joshuag1000/Raspberry-Pi-Pico-MacroPad/blob/main/pico_rgb_keypad/Main-Code.cpp
I need the checks at lines 158+

**Reply by chegewara** [reply]
You need this callback and reverse engineer values reported.
```
void tud_hid_set_report_cb(uint8_t report_id, hid_report_type_t report_type, uint8_t const* buffer, uint16_t bufsize)
```

**Comment by hathach** [comment]
thanks @chegewara for pointing it you, whenever the host get the lock changes, it will set all the HID keyboard with appropriate value. Which will be eventually trigger by  `tud_hid_set_report_cb()`, just parsed it out. TODO here is the hint, I will try to implement it in stock example just for demonstration 

https://github.com/hathach/tinyusb/blob/master/examples/device/hid_composite/src/main.c#L258
